modules/storage/src/utils/index.ts

```diff
@@ -62,3 +67,176 @@ export async function deepPathHandler(
+  }
+  const url = file.isPublic
+    ? await storageProvider
+        .container(container)
+        .getPublicUrl((folder === '/' ? '' : folder) + name)
+    : null;
+  const updatedFile = (await File.getInstance().findByIdAndUpdate(file._id, {
+    name,
+    alias,
+    folder,
+    container,
+    url,
+    mimeType,
+  })) as File;
+  updateFileMetrics(file.size, (data as Buffer).byteLength);
+  return updatedFile;
+}
+
+export async function _updateFileUploadUrl(
+  storageProvider: IStorageProvider,
+  file: File,
+  params: IFileParams,
+): Promise<{ file: File; url: string }> {
+  const { name, alias, folder, container, mimeType, size } = params;
+  let updatedFile;
+  const onlyDataUpdate =
+    name === file.name && folder === file.folder && container === file.container;
+  if (onlyDataUpdate) {
+    updatedFile = await File.getInstance().findByIdAndUpdate(file._id, {
+      mimeType,
+      alias,
+      ...{ size: size ?? file.size },
+    });
+  } else {
+    await storageProvider
+      .container(container)
+      .store(
+        (folder === '/' ? '' : folder) + name,
+        Buffer.from('PENDING UPLOAD'),
+        file.isPublic,
+      );
+    await storageProvider
+      .container(file.container)
+      .delete((file.folder === '/' ? '' : file.folder) + file.name);
+    const url = file.isPublic
+      ? await storageProvider
+          .container(container)
+          .getPublicUrl((folder === '/' ? '' : folder) + name)
+      : null;
+    updatedFile = await File.getInstance().findByIdAndUpdate(file._id, {
+      name,
+      alias,
+      folder,
+      container,
+      url,
+      mimeType,
+      ...{ size: size ?? file.size },
+    });
+  }
+  if (!isNil(size)) updateFileMetrics(file.size, size!);
+  const uploadUrl = (await storageProvider
+    .container(container)
+    .getUploadUrl((folder === '/' ? '' : folder) + name)) as string;
+  return { file: updatedFile!, url: uploadUrl };
+}
+
+export function updateFileMetrics(currentSize: number, newSize: number) {
+  const fileSizeDiff = Math.abs(currentSize - newSize);
+  fileSizeDiff < 0
+    ? ConduitGrpcSdk.Metrics?.increment('storage_size_bytes_total', fileSizeDiff)
+    : ConduitGrpcSdk.Metrics?.decrement('storage_size_bytes_total', fileSizeDiff);
+}
+
+export async function validateName(
+  name: string | undefined,
+  folder: string,
+  container: string,
+) {
+  if (!name) {
+    return randomUUID();
+  }
+  const config = ConfigController.getInstance().config;
+  const escapedName = name.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&');
+  const count = await File.getInstance().countDocuments({
+    $and: [
+      { $or: [{ name }, { name: { $regex: `^${escapedName} \\(\\d+\\)` } }] },
+      { folder: folder },
+      { container: container },
+    ],
+  });
+  if (count === 0) {
+    return name;
+  } else if (!config.suffixOnNameConflict) {
+    throw new GrpcError(status.ALREADY_EXISTS, 'File already exists');
+  } else {
+    return `${name} (${count})`;
```

Comment:
count needs to be added before file type, if file type exists in name. For example on file myFile.pdf it should become myFile (1).pdf. While myFile, should become myFile (1).